ocaml/xapi/xapi_vdi.ml

```diff
@@ -25,6 +25,30 @@ open Printf
 (**************************************************************************************)
 (* current/allowed operations checking                                                *)
 
+let check_sm_feature_error op sm_features sr =
```

Comment:
In this sort of case I'm not so convinced.

Since `check_sm_feature_error` is called from only one place it could be inlined at its call site, but I made it a separate function to keep `check_operation_error` more readable.

The `check_operation_error` function is still too long to fit on my screen, and I think it would be less readable if it were to contain the implementation of `check_sm_feature_error` (whether as an inner function or as a inline chunk of code).